Source/Core/VideoCommon/CommandProcessor.cpp

```diff
@@ -221,30 +221,30 @@ void CommandProcessorManager::RegisterMMIO(Core::System& system, MMIO::Mapping*
                    MMIO::InvalidWrite<u16>());
   }
 
-  mmio->Register(base | STATUS_REGISTER, MMIO::ComplexRead<u16>([](Core::System& system, u32) {
-                   auto& cp = system.GetCommandProcessor();
-                   system.GetFifo().SyncGPUForRegisterAccess(system);
-                   cp.SetCpStatusRegister(system);
+  mmio->Register(base | STATUS_REGISTER, MMIO::ComplexRead<u16>([](Core::System& system_, u32) {
+                   auto& cp = system_.GetCommandProcessor();
+                   system_.GetFifo().SyncGPUForRegisterAccess(system_);
```

Comment:
Eww... Can we rename the `system` in the parent scope instead? It's only used once.

Reply:
BTW, is there any good reason to actually pass in system to the lambda's via the parameter here?

Since we re-create these MMIO handlers every boot (and they are per system), a capture is also non-global and will produce identical results. Or is there a long-term plan to only initialise these lambda's only once (presumably at compile time?)

Reply:
I asked about this back in https://github.com/dolphin-emu/dolphin/pull/11166 and didn't get a whole lot of replies, but IMO it's cleaner to pass the system to the MMIO handler than to have every single MMIO handler capture it. But ymmv.

Reply:
Thinking about it from a high-level design perspective, yes, it's cleaner to pass system to the MMIO handler. 

Mostly because it makes it more clear that MMIO handlers consumes system, rather than hiding it away in a lambda. And as I said, it does open up some possibilities for a more static boot process. 

But that's not going to stop me also getting annoyed that it's forcing us to add underscores to variable names to avoid shadow warnings.